app/samples/sample-common-test/src/main/java/com/sygic/samples/idling/BaseIdlingResource.kt

```diff
@@ -0,0 +1,13 @@
+package com.sygic.samples.idling
+
+import androidx.test.espresso.IdlingResource
+import com.sygic.samples.CommonSampleActivity
+
+abstract class BaseIdlingResource(protected val activity: CommonSampleActivity) : IdlingResource {
+
+    protected var callback: IdlingResource.ResourceCallback? = null
+
+    override fun registerIdleTransitionCallback(callback: IdlingResource.ResourceCallback?) {
```

Comment:
preco overridujes tento register?

Reply:
vlastní callback? nerozumím. však v tom registerIdleTransitionCallback přijde ta instance callback-u, ten stejný se uloží a později provolá. žádný vlastní tu nevidím.

Reply:
no ano, ze nikde ten `callback` nevolas. Pozeram ze `IdlingResource` je interface, takze ok, `registerIdleTransitionCallback` super nema. Takze predpokladam, ze to je na podedenej triede, aby ten callback zavolala? Asi by som sa na to nespoliehal a trosku inak nadizajnoval tu nadtriedu.

nieco ako 

```
override final fun isIdleNow(): Boolean {
    if (checkIdle()) {    //abstract
        callback.onTransitionToIdle()
     }
}
```